packages/wallets/src/services/wallet-factory.ts

```diff
@@ -1,38 +1,61 @@
+import { PublicKey } from "@solana/web3.js";
 import type { Address } from "viem";
+
 import { SolanaMPCWallet } from "../solana";
 import type { WalletTypeToArgs, WalletTypeToWallet } from "./types";
 import type { ApiClient, CreateWalletResponse } from "../api";
 import { EVMSmartWallet } from "../evm";
 import { SolanaSmartWallet } from "../solana";
-import { PublicKey } from "@solana/web3.js";
 import { parseSolanaSignerInput } from "../solana/types/signers";
+import type { WalletOptions } from "../utils/options";
 
 export class WalletFactory {
     constructor(private readonly apiClient: ApiClient) {}
 
     public async getOrCreateWallet<WalletType extends keyof WalletTypeToArgs>(
         type: WalletType,
-        args: WalletTypeToArgs[WalletType]
+        args: WalletTypeToArgs[WalletType],
+        options?: WalletOptions
+    ): Promise<WalletTypeToWallet[WalletType]> {
+        try {
+            const walletResponse = await this.createWallet(type, args, options);
+            const wallet = this.createWalletInstance(type, walletResponse, args, options);
+            await options?.experimental_callbacks?.onWalletCreationComplete?.(wallet);
```

Comment:
feels weird to me to have `onWalletCreationStart` in one method, and the rest here

Reply:
yeah I agree https://github.com/Crossmint/crossmint-sdk/pull/985/commits/e72d66c8275c0340e6f6c96f47c9ca02b8f8e2f5